torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4710,8 +4749,15 @@ def issubset(x, y):
                 # Try to invert the equality
                 r = try_solve(sympy.Eq(a, tgt), b, floordiv_inequality=False)
                 if r is not None:
-                    b_bound = self.bound_sympy(r[1])
-                    self.var_to_range[b] = b_bound & self.var_to_range[b]
+                    self.log.debug("set_replacement: solve for %s in %s == %s gives %s", b, a, tgt, r)
+                    # The solution here can be non-integral, for example, if
+                    # we have s0 = 2*s1, then s1 = s0/2.  What we would like
+                    # to do is calculated the bounds in arbitrary precision,
+                    # and then requantize the bound to integers when we are
+                    # done.
+                    rat_b_bound = self.bound_sympy(r[1])
+                    b_bound = ValueRanges(CeilToInt(rat_b_bound.lower), FloorToInt(rat_b_bound.upper))
+                    self._update_var_to_range(b, b_bound)
```

Comment:
These are the new changes from the itnernal revert. It turned out that the old logic (that discarded solutions here which were not integral) impeded replacements entirely when you had a nontrivial value range on the symbol being substituted. So I reallow rational intermediate values and round them at the very end (since fractional solutions cannot be real entries in the value range). This also works because I deleted all of the is integer asserts in our custom functions.

More discussion at https://github.com/pytorch/pytorch/issues/128053